pandas/io/json/_normalize.py

```diff
@@ -228,14 +228,17 @@ def json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js, spec):
-        result = js
+    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> Iterable:
+        result: Any = js
```

Comment:
Is this still necessary?

Reply:
What part? Stronger typing is always preferred I guess? That was suggested by @jreback . If I remove the `Any` I get

```
pandas/io/json/_normalize.py:240: error: Incompatible types in assignment (expression has type "List[<nothing>]", variable has type "Dict[str, Any]")
```

Reply:
Yea typing as `Any` isn't strong and for practical purposes would be the same as no typing.

`result: Union[List, Dict[str, Any]]` would be preferable for now; can refine later

Reply:
Wont work unfortunately. See https://github.com/python/mypy/issues/2220

```
pandas/io/json/_normalize.py:238: error: No overload variant of "__getitem__" of "list" matches argument type "str"
pandas/io/json/_normalize.py:238: note: Possible overload variants:
pandas/io/json/_normalize.py:238: note:     def __getitem__(self, int) -> Any
pandas/io/json/_normalize.py:238: note:     def __getitem__(self, slice) -> List[Any]
```

Adding `str` gives
```
error: Invalid index type "str" for "Union[str, List[Any], Dict[str, Any]]"; expected type "Union[int, slice]"
```

Reply:
If you don't mind I'll keep it at `Any` as that also benefits readability (imho)

Reply:
Yea the way this variable is reused makes things kind of tricky; not from anything you have here just a historical artifact

Instead of `: Any` can you just add a `# type: ignore` comment? The former we mostly use internally when an object really can be anything. That isn't true here, but just need some refactor to get it to work cleanly in a follow up and the ignore would be more indicative for that

Reply:
Sure